priv/templates/phx.gen.auth/reset_password_live_test.exs

```diff
@@ -0,0 +1,166 @@
+    end
+
+    test "does not send reset password token if email is invalid", %{conn: conn} do
+      {:ok, lv, _html} = live(conn, Routes.<%= schema.route_helper %>_reset_password_path(conn, :new))
+
+      {:ok, conn} =
+        lv
+        |> form("#reset_password_form", <%= schema.singular %>: %{"email" => "unknown@example.com"})
+        |> render_submit()
+        |> follow_redirect(conn, "/")
+
+      assert get_flash(conn, :info) =~ "If your email is in our system"
+      assert Repo.all(<%= inspect context.alias %>.<%= inspect schema.alias %>Token) == []
+    end
+  end
+
+  describe "Reset password(new password page)" do
+    setup do
+      <%= schema.singular %> = <%= schema.singular %>_fixture()
+
+      token =
+        extract_<%= schema.singular %>_token(fn url ->
+          Accounts.deliver_<%= schema.singular %>_reset_password_instructions(<%= schema.singular %>, url)
+        end)
+
+      %{token: token}
+    end
+
+    test "renders reset password with valid token", %{conn: conn, token: token} do
+      {:ok, _lv, html} = live(conn, Routes.<%= schema.route_helper %>_reset_password_path(conn, :edit, token))
+
+      assert html =~ "<h1>Reset password</h1>"
+    end
+
+    test "does not render reset password with invalid token", %{conn: conn} do
+      {:error, {:redirect, redirect}} =
+        live(conn, Routes.<%= schema.route_helper %>_reset_password_path(conn, :edit, "invalid"))
+
+      assert redirect == %{
+               flash: %{"error" => "Reset password link is invalid or it has expired."},
+               to: "/"
+             }
+    end
+  end
+
+  describe "Reset password" do
+    setup do
+      <%= schema.singular %> = <%= schema.singular %>_fixture()
+
+      token =
+        extract_<%= schema.singular %>_token(fn url ->
+          <%= inspect context.alias %>.deliver_<%= schema.singular %>_reset_password_instructions(<%= schema.singular %>, url)
+        end)
+
+      %{token: token, <%= schema.singular %>: <%= schema.singular %>}
+    end
+
+    test "resets password once", %{conn: conn, token: token, <%= schema.singular %>: <%= schema.singular %>} do
+      {:ok, lv, _html} = live(conn, Routes.<%= schema.route_helper %>_reset_password_path(conn, :edit, token))
+
+      {:ok, conn} =
+        lv
+        |> form("#reset_password_form",
+          <%= schema.singular %>: %{
+            "password" => "new valid password",
+            "password_confirmation" => "new valid password"
+          }
+        )
+        |> render_submit()
+        |> follow_redirect(conn, Routes.<%= schema.singular %>_login_path(conn, :new))
+
+      refute get_session(conn, :<%= schema.singular %>_token)
+      assert get_flash(conn, :info) =~ "Password reset successfully"
+      assert <%= inspect context.alias %>.get_<%= schema.singular %>_email_and_password(<%= schema.singular %>.email, "new valid password")
```

Comment:
```suggestion
      assert <%= inspect context.alias %>.get_<%= schema.singular %>_by_email_and_password(<%= schema.singular %>.email, "new valid password")
```

Reply:
nice catch!
I generated the auth modules in a test app and it never raised an error 🤔

Reply:
Ah! now I see.